### ISSUE TITLE
MacOS: option panel overflow

### ISSUE THREAD

**Issue by ding0z** [vkbo/novelWriter#989]
While changing the default font, I've noticed that the option panel opens without full screen but without respecting the actual size of the screen. It is actually much wider. I have to resize it every time and pressing 'ok' doesn't seem to save the size of the panel.

This issue occurs on laptop with or without external monitor connected.

**Comment by vkbo** [comment]
Could you post a screenshot? I'm not sure I understand the issue.

There are some issues related to DPI scaling. The support for high DPI screens is a bit quirky in Qt5 (see #864). It is fixed in Qt6. There is some additional code in novelWriter for managing it, but it isn't perfect.

**Comment by ding0z** [comment]
It is related to the Preferences panel wider than the screen.
Even the font options dialog box becomes too small if I increase the font size too much, but that's easily fixable. Courier/Arial 14 for that works just fine.

**Comment by vkbo** [comment]
There may be a bug related to detecting UI scale on macOS. Could you start novelWriter from the command line with the `--verbose` flag? There should be a line in the verbose output stating the DPI and UI scale. The two lines should come fairly close to the top and state "VERBOSE  GUI DPI:" and "VERBOSE  GUI Scale:" and two numbers.

**Comment by ding0z** [comment]
Here they are:

`[2022-02-08 16:09:59,926]           theme.py:142   VERBOSE   GUI DPI: 72.0
[2022-02-08 16:09:59,926]           theme.py:143   VERBOSE   GUI Scale: 0.75
`

**Comment by vkbo** [comment]
You can see yourself how messy all of this is in Qt5: https://doc.qt.io/qt-5/highdpi.html

I have an issue up to review the implementation (#864). I'll bump it up in my ToDo list. The problem is that there are various solutions depending on which minor version of Qt5 you're on, and a macOS solution that requires the plist file to be present. Since novelWriter isn't properly bundled for macOS, this file does not currently exist.

Since Qt5 is no longer under development, and Qt 5.15 is the LTS release, I could probably further restrict the minimum vrsion requirement to 5.6 and see if I can find a consistent way of handling DPI scaling. Apparently all of this is simpler in Qt6.

**Comment by ding0z** [comment]
Sure. Not a bit problem anyway. A double click to the title bar brings everything back to normal.

**Comment by vkbo** [comment]
I discovered a potential bug in the preferences and project settings dialogs that may have caused this. Is the issue still present in the current main branch?

**Comment by ding0z** [comment]
Yes, still present. (v1.6.4)

**Comment by vkbo** [comment]
No, I mean the current source code. But never mind, I think I spotted the actual issue now, and will fix it for the next release. I may do a 1.6.5 as well since there are a couple of minor bugs that need fixing.

**Comment by ding0z** [comment]
Ah, I understand. Ok, I'll wait and report in case once the 1.6.5 is available.

**Comment by vkbo** [comment]
Fixed and released as 1.6.5.
